src/fast_check/mod.rs

```diff
@@ -56,6 +56,36 @@ impl std::fmt::Debug for FastCheckDiagnosticRange {
   }
 }
 
+impl PartialEq for FastCheckDiagnosticRange {
+  fn eq(&self, other: &Self) -> bool {
+    let self_range_start =
+      self.text_info.line_and_column_index(self.range.start);
```

Comment:
Why does it need to compute the line and column index? Can it just check if the range starts are equal?

Reply:
I guess so - I was somehow thinking that the diagnostic ranges could come from different parses, so the SourcePos would not line up

Reply:
Yeah, I think we can just compare the ranges directly. If they come from different parses they should still be equal.